src/libcore/sync/atomic.rs

```diff
@@ -2518,6 +2547,7 @@ pub fn fence(order: Ordering) {
 #[inline]
 #[stable(feature = "compiler_fences", since = "1.21.0")]
 pub fn compiler_fence(order: Ordering) {
+    // SAFETY: doesn't compile to machine code
     unsafe {
         match order {
             Acquire => intrinsics::atomic_singlethreadfence_acq(),
```

Comment:
are any of these intrinsics unsafe for a reason? Invoking the `compiler_fence` function is safe, so the intrinsics could be safe

Reply:
Found this:

> As with any other FFI functions, these are always unsafe to call.

https://doc.rust-lang.org/1.1.0/book/intrinsics.html

The docs are old but I think they're still true. Meaning, I don't think there's a way around it unless you add a wrapper function but I'm not sure that adds anything.

Reply:
We have safe intrinsics these days, like `size_of`.